instrumentation/build.gradle.kts

```diff
@@ -18,19 +20,29 @@ subprojects {
       testCompileOnly(project(path = ":testing:armeria-shaded-for-testing", configuration = "shadow"))
     }
   }
-}
 
-tasks {
-  register("listInstrumentations") {
-    group = "Help"
-    description = "List all available instrumentation modules"
-    doFirst {
-      File("instrumentation-list.txt").printWriter().use { out ->
-        subprojects
-          .filter { it.plugins.hasPlugin("io.opentelemetry.instrumentation.muzzle-check") }
-          .map { it.path }
-          .forEach { out.println(it) }
-      }
-    }
+  plugins.withId("io.opentelemetry.instrumentation.muzzle-check") {
+    muzzleTasks.add(subProj.tasks.named("muzzle").get())
   }
 }
+
+// TODO (trask) how to know that muzzleTasks populated above before used below?
```

Comment:
If the muzzle tasks are actually being run then maybe it's OK but indeed this pattern doesn't look quite reliable. I don't know if it's important for us to split the list into `1111222233334444`, if `123412341234` is OK I would create the tasks first and then in the `plugins.withId` callback `AtomicInteger.getAndIncrement() % 4` to pick the task to register into.

Alternatively we would need to use `evaluationDependsOn` and `afterEvaluate`, but reducing evaluation order dependencies tends to help understand the build.

Reply:
> if `123412341234` is OK I would create the tasks first and then in the `plugins.withId` callback `AtomicInteger.getAndIncrement() % muzzleTasks.size()` to pick the task to register into.

ya, I like that, do you know if the subprojects ordering is guaranteed to be the same across multiple invocations (on same git sha)?

Reply:
As far as I know Gradle project evaluation order is deterministic (a BFS) which would make this deterministic too, though I can't say with 100% confidence without trying

Reply:
hm, this worries me: https://www.google.com/search?q=gradle+%22the+task+execution+order+is+not+deterministic%22

Reply:
Here we're relying on project evaluation order, not task evaluation order. Luckily, project evaluation order is defined by Gradle

https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#N14CB4